epoxy-adapter/src/main/java/com/airbnb/epoxy/ChangeAnimationDelegator.java

```diff
@@ -0,0 +1,168 @@
+package com.airbnb.epoxy;
+
+import android.animation.AnimatorListenerAdapter;
+import android.animation.EpoxyItemAnimation.AnimatorListener;
+import android.support.annotation.NonNull;
+import android.support.v7.widget.RecyclerView.ItemAnimator;
+import android.support.v7.widget.RecyclerView.ViewHolder;
+
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * ItemAnimator that wraps any other ItemAnimator and delegates change animations to
+ * {@link EpoxyModel}
+ */
+public final class ChangeAnimationDelegator extends ItemAnimatorDecorator
+    implements AnimatorListener {
+  /* use a BiMap so we can use the same listener for every animator and quickly find the holder for
+     an animation. We do this to be able to remove only OUR listener from an animator and not those
+     that the user might have set */
+  @NonNull private final BiHashMap<ViewHolder, EpoxyItemAnimation> mChangeAnimations =
+      new BiHashMap<>();
+
+  public ChangeAnimationDelegator(@NonNull ItemAnimator decoratedAnimator) {
+    super(decoratedAnimator);
+  }
+
+  public long getChangeDuration() {
+    long maxDuration = super.getChangeDuration();
+
+    for (EpoxyItemAnimation anim : mChangeAnimations.values()) {
+      maxDuration = Math.max(maxDuration, anim.getDuration());
+    }
+
+    return maxDuration;
+  }
+
+  public void onAnimationEnd(EpoxyItemAnimation anim) {
+    dispatchAnimationFinished(mChangeAnimations.inverse().get(anim));
+    mChangeAnimations.inverse().remove(anim);
+
+    if (mChangeAnimations.isEmpty()) {
+      dispatchChangeAnimationsFinished();
+    }
+  }
+
+  @Override
+  public void onAnimationCancel(EpoxyItemAnimation anim) {
+    mChangeAnimations.inverse().remove(anim);
+  }
+
+  public void onAnimationStart(EpoxyItemAnimation anim) {
+    dispatchAnimationStarted(mChangeAnimations.inverse().get(anim));
+  }
+
+  @SuppressWarnings("unchecked")
+  public boolean animateChange(
+      @NonNull final ViewHolder oldHolder, @NonNull final ViewHolder newHolder,
+      @NonNull ItemHolderInfo preLayoutInfo, @NonNull ItemHolderInfo postLayoutInfo) {
+
+    if (oldHolder == newHolder && oldHolder instanceof EpoxyViewHolder
+        && canHolderAnimateChanges((EpoxyViewHolder) oldHolder)) {
+
+      final EpoxyViewHolder holder = ((EpoxyViewHolder) oldHolder);
+      final EpoxyModel newModel = holder.getModel();
+      final Object viewToBind = holder.objectToBind();
+      final EpoxyModel oldModel = DiffPayload.getModelFromPayload();
+
+      final EpoxyItemAnimation oldAnim = mChangeAnimations.get(oldHolder);
+
+      if (oldAnim == null || !oldAnim.isRunning()) {
+        EpoxyItemAnimation newAnim = newModel.bindChangesAnimated(viewToBind, oldModel);
+        newAnim.addListener(this);
+        mChangeAnimations.put(holder, newAnim);
+      } else {
+        EpoxyItemAnimation continuedAnim =
```

Comment:
yeah, mainly just by checking `animation.isStarted()`. Mostly I'm trying to think "what checks can we do to make sure the user is doing what they should", and since we are starting the animation ourselves, this seems like an easy one to validate. Any other ideas on things we can check to make sure the user sets up the animation properly?